### PR TITLE
Persistent storage

### PR DESCRIPTION
**Pull request by Rebal67**
issue 25

### CODE REVIEW

File: src/initialize.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+/**
+ * initliaze the sdk and return the user object
+ * @param keyManager  the key manager
+ * @param storage  the storage
+ * @returns the user object
+ */
+function initialize(keyManager: KeyManager, storage: PersistantStorage): User {
+  var _storage, _keyManager;
```
Comment (theblockstalk) [inline src/initialize.ts:13]: what is the purpose of L13-19? I don't think it adds anything.

File: src/storage.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   */
+  store(key: string, value: any): void;
+
+  /**
+   * @param key - The key to retrieve the data from
+   * @returns The data stored under the key
+   * @throws {Error} If the data could not be retrieved
+   */
+  retrieve(key: string): any;
+}
```
Comment (theblockstalk) [inline src/storage.ts:17]: this also should define remove(key: string) and maybe clear()

File: src/initialize.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { User } from "./user";
+
+
+/**
+ * initliaze the sdk and return the user object
+ * @param keyManager  the key manager
+ * @param storage  the storage
+ * @returns the user object
+ */
+function initialize(keyManager: KeyManager, storage: PersistantStorage): User {
```
[the remark this answers is not in the document]
Comment (Rebal67) [reply src/initialize.ts:12]: i named it like that so it initiate the library meaning the storage will be injected in multiple places in the future

File: src/storage.ts (hunk 2 above)
Comment (Rebal67) [reply src/storage.ts:17]: clear() implmented in logout pull request

File: src/initialize.ts (hunk 1 above)
Comment (theblockstalk) [reply src/initialize.ts:13]: do we do that? I though we always went through `userObj.storage.x`...

Comment (Rebal67) [reply src/initialize.ts:13]: right ill remove it

**Pull request merged**